specification/appconfiguration/AppConfiguration/routes.tsp

```diff
@@ -298,17 +324,26 @@ op createSnapshot is Foundations.LongRunningOperation<
   Error
 >;
 
+/** The content types for updateSnapshot. */
+#suppress "@azure-tools/typespec-azure-core/no-closed-literal-union" "Needs fix: https://github.com/microsoft/typespec/issues/2853"
+union UpdateSnapshotContentType {
+  /** The content type for a JSON merge patch operation. */
+  "application/merge-patch+json",
+
+  /** The content type for a JSON object. */
+  "application/json",
+}
+
 #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Pre-existing API contract"
 @summary("Updates the state of a key-value snapshot.")
 @doc("Updates the state of a key-value snapshot.")
 @route("/snapshots/{name}")
 @patch(#{ implicitOptionality: true })
 op updateSnapshot is AppConfigOperation<
   {
-    #suppress "@azure-tools/typespec-azure-core/no-closed-literal-union" "Needs fix: https://github.com/microsoft/typespec/issues/2853"
     @doc("Content-Type header")
     @header("Content-Type")
-    contentType: "application/merge-patch+json" | "application/json";
+    contentType: UpdateSnapshotContentType;
```

Comment:
those changes do create a few new types in teh swagger too, what was the previous behavior of tcgc, does it really need to be a named model